src/lib/DefaultRuleManager.ts

```diff
@@ -112,8 +112,8 @@ export class DefaultRuleManager implements RuleManager {
 	protected async resolveEngineFilters(filters: RuleFilter[], engineOptions: Map<string,string> = new Map()): Promise<RuleEngine[]> {
 		let filteredEngineNames: readonly string[] = null;
 		for (const filter of filters) {
-			if (filter.filterType === FilterType.ENGINE) {
-				filteredEngineNames = filter.filterValues;
+			if (isEngineFilter(filter)) {
```

Comment:
Using typeguards that Josh pointed out https://www.typescriptlang.org/docs/handbook/advanced-types.html

Reply:
Thanks for the link - very helpful to read and understand about it.